### PR TITLE
[RUBY-1522] Update to Rails 7

### PR DESCRIPTION
**Pull request by srushe**
Ticket: [RUBY-1522](https://contrast.atlassian.net/browse/RUBY-1522)

Adds rails 7 to the pipeline, updating the existing rails from 6.0.4.1 to 7.0.1. Assuming this meets with approval I'll then merge it and rebuild the docker images from it.

### CODE REVIEW

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
   linting:
     name: "Running ${{ matrix.linter }}"
     runs-on: ubuntu-latest
-    container: ruby:2.5
+    container: ruby:2.7
```
Comment (srushe) [inline .github/workflows/pipeline.yml:17]: This change was required in order to stop bundler complaining with the decidedly unhelpful message "Your bundle is locked to rails (7.0.1) from rubygems repository [rubygems.org](http://rubygems.org/) or installed locally, but that version can no longer be found in that source. That means the author of rails (7.0.1) has removed it."

File: Gemfile.lock [hunk 2]
```diff
@@ -11,89 +11,105 @@ PATH
   specs:
     vulneruby_engine (0.1.0)
       grape (~> 1.5.3)
-      rails (~> 6.0.3, >= 6.0.3.1)
-      rake (~> 12.0)
+      rake (~> 12.3.0)
+      sidekiq (~> 6.3, >= 6.3.1)
+      sidekiq-status (~> 2.1)
       sinatra (~> 2.0)
+      sprockets-rails
```
Comment (srushe) [inline Gemfile.lock:18]: `sprocket-rails` was removed as a default gem in Rails 7, hence the addition of this explicitly.

File: config/initializers/secrets.rb [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 # Generates a few 'secrets' meant to represent API keys or something similar
 
+require 'vulneruby_engine/application_record'
+require 'vulneruby_engine/secret'
```
Comment (srushe) [inline config/initializers/secrets.rb:6]: Explicit requires added due to the following deprecation warning now being issued regarding autoloading during initialisation.

```
DEPRECATION WARNING: Initialization autoloaded the constants VulnerubyEngine::ApplicationRecord, VulnerubyEngine::Secret.

Being able to do this is deprecated. Autoloading during initialization is going
to be an error condition in future versions of Rails.
```

File: spec/dummy/config/application.rb [hunk 4]
```diff
@@ -3,6 +3,7 @@
 require_relative('boot')
 
 require('rails/all')
+require('sprockets/railtie')
```
Comment (srushe) [inline spec/dummy/config/application.rb:6]: Require now that sprockets isn't included by default in Rails 7.

File: spec/dummy/config/initializers/filter_parameter_logging.rb [hunk 5]
```diff
@@ -3,4 +3,6 @@
 # Be sure to restart your server when you modify this file.
 
 # Configure sensitive parameters which will be filtered from the log file.
-Rails.application.config.filter_parameters += [:password]
+Rails.application.config.filter_parameters += [
+  :password, :passw, :secret, :token, :_key, :crypt, :salt, :certificate, :otp, :ssn
+]
```
Comment (srushe) [inline spec/dummy/config/initializers/filter_parameter_logging.rb:8]: The new default for this for 7, other than `:password` which I carried over from our existing filter.

**Review by kaloyan-mm — APPROVED** [review]

**Pull request merged**